### PR TITLE
feat: return status on available locales

### PR DESCRIPTION
**Pull request by Marc-Roig**
Makes sure we follow the following contract for document metadata:

```
type DocumentMetadata = {
  // All status of the returned locale
  availableStatus: Pick<Document, 'id' | AT_FIELDS  | BY_FIELDS>[];
  // Available locales within the same status of the returned document
  availableLocales: Pick<Document, 'id' | 'locale' | AT_FIELDS | 'status'>[];
};
```

Before we were not returning the AT_FIELDS, BY_FIELDS, neither the status on available locales.

Also, a published version will return it's status as "modified" if the draft version has modifications. Before this was not the case and a published document alwyas had "published" as its status

Resolves:
CONTENT-2220
CONTENT-2216

### CODE REVIEW

**Comment by Marc-Roig** [comment]
I am working on some api tests for the document metadata

File: packages/core/core/src/services/document-service/transform/fields.ts [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
+import { Documents, Common } from '@strapi/types';
+
+type Fields = Documents.Params.Pick<Common.UID.Schema, 'fields'>['fields'];
+
+export const transformFields = (fields: Fields): Fields => {
+  // If  it's a string, and it doesn't contain documentId, should be an array
+  // '*' => '*'
+  // 'documentId' => 'documentId' , 'documentId,name' => 'documentId,name'
+  // 'name' => ['name', 'documentId']
+  // 'id' => 'documentId'
```
Comment (Marc-Roig) [inline packages/core/core/src/services/document-service/transform/fields.ts:10]: @jhoward1994 unrelated to this pr but realised there were a couple of cases we were not covering here!

Comment (jhoward1994) [reply packages/core/core/src/services/document-service/transform/fields.ts:10]: Thank you!

File: packages/core/content-manager/server/src/services/document-metadata.ts [hunk 2]
```diff
@@ -56,45 +69,53 @@ export default ({ strapi }: { strapi: Strapi }) => ({
...
+      };
+    });
   },
 
+  /**
+   * Returns available status of a document for the current locale
+   */
+  async getAvailableStatus(version: DocumentVersion, allVersions: DocumentVersion[]) {
+    // Find the other status of the document
+    const status = version.publishedAt !== null ? 'draft' : 'published';
```
Comment (jhoward1994) [inline packages/core/content-manager/server/src/services/document-metadata.ts:98]:
```suggestion
    const status = version.publishedAt !== null ? CONTENT_MANAGER_STATUS.DRAFT : CONTENT_MANAGER_STATUS.PUBLISHED;
```

?

Comment (Marc-Roig) [reply packages/core/content-manager/server/src/services/document-metadata.ts]: done!

File: packages/core/content-manager/server/src/services/document-metadata.ts [hunk 3]
```diff
@@ -56,45 +69,56 @@ export default ({ strapi }: { strapi: Strapi }) => ({
...
+
+    // Get version that match the current locale and not match the current status
+    const availableStatus = allVersions.find((v) => {
+      const matchLocale = v.locale === version.locale;
+      const matchStatus = status === 'published' ? v.publishedAt !== null : v.publishedAt === null;
+      return matchLocale && matchStatus;
+    });
+
+    if (!availableStatus) return availableStatus;
+
```
Comment (jhoward1994) [inline packages/core/content-manager/server/src/services/document-metadata.ts:111]:
```suggestion
    return { ...pick(AVAILABLE_STATUS_FIELDS, availableStatus), status };
```

**Review by jhoward1994 — APPROVED** [review]

File: packages/core/content-manager/server/src/services/document-metadata.ts (hunk 3 above)
Comment (Marc-Roig) [reply packages/core/content-manager/server/src/services/document-metadata.ts:111]: indeed, will add this in another PR, as I wanted to discuss something with josh about this!

**Pull request merged**